server/rosetta/services/construction.go

```diff
@@ -0,0 +1,318 @@
+package services
+
+import (
+	"context"
+	"crypto/sha256"
+	"encoding/hex"
+	"strings"
+
+	"github.com/cosmos/cosmos-sdk/client"
+
+	"github.com/btcsuite/btcd/btcec"
+	"github.com/coinbase/rosetta-sdk-go/types"
+	crg "github.com/tendermint/cosmos-rosetta-gateway/rosetta"
+	"github.com/tendermint/tendermint/crypto"
+
+	"github.com/cosmos/cosmos-sdk/client/tx"
+	"github.com/cosmos/cosmos-sdk/crypto/keys/secp256k1"
+	"github.com/cosmos/cosmos-sdk/server/rosetta"
+	"github.com/cosmos/cosmos-sdk/server/rosetta/cosmos/conversion"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/tx/signing"
+	authsigning "github.com/cosmos/cosmos-sdk/x/auth/signing"
+	bank "github.com/cosmos/cosmos-sdk/x/bank/types"
+)
+
+// interface implementation assertion
+var _ crg.ConstructionAPI = OnlineNetwork{}
+
+func (on OnlineNetwork) ConstructionCombine(ctx context.Context, request *types.ConstructionCombineRequest) (*types.ConstructionCombineResponse, *types.Error) {
+	txBldr, err := on.getTxBuilderFromBytesTx(request.UnsignedTransaction)
+	if err != nil {
+		return nil, rosetta.ToRosettaError(err)
+	}
+
+	var sigs = make([]signing.SignatureV2, len(request.Signatures))
+	for i, signature := range request.Signatures {
+		if signature.PublicKey.CurveType != "secp256k1" {
+			return nil, rosetta.ErrUnsupportedCurve.RosettaError()
+		}
+
+		cmp, err := btcec.ParsePubKey(signature.PublicKey.Bytes, btcec.S256())
+		if err != nil {
+			return nil, rosetta.ToRosettaError(err)
+		}
+
+		compressedPublicKey := make([]byte, secp256k1.PubKeySize)
+		copy(compressedPublicKey, cmp.SerializeCompressed())
+		pubKey := &secp256k1.PubKey{Key: compressedPublicKey}
+
+		accountInfo, err := on.client.AccountInfo(ctx, sdk.AccAddress(pubKey.Address()).String(), nil)
+		if err != nil {
+			return nil, rosetta.ToRosettaError(err)
+		}
+
+		sig := signing.SignatureV2{
+			PubKey: pubKey,
+			Data: &signing.SingleSignatureData{
+				SignMode:  signing.SignMode_SIGN_MODE_LEGACY_AMINO_JSON,
```

Comment:
why using Legacy Amino?

Reply:
I think it relates to the issue that we raised a couple of weeks ago. SIGN_MODE_DIRECT can't be used

Reply:
yes, we couldn't use SIGN_MODE_DIRECT as it requires to SetSignatures with pubkeys which rosetta-cli doesn't support atm https://github.com/cosmos/cosmos-sdk/blob/90e9370bd80d9a3d41f7203ddb71166865561569/client/tx/tx.go#L404-L411